common/web/input-processor/src/text/inputProcessor.ts

```diff
@@ -107,8 +107,20 @@ export default class InputProcessor {
       if(keyEvent.baseTranscriptionToken) {
         const transcription = this.contextCache.get(keyEvent.baseTranscriptionToken);
         if(transcription) {
-          // Restores full context, including deadkeys in their exact pre-keystroke state.
-          outputTarget.restoreTo(transcription.preInput);
+          // Has there been a context change at any point during the multitap?  If so, we need
+          // to revert it.  If not, we assume it's a layer-change multitap, in which case
+          // no such reset is needed.
+          if(!isEmptyTransform(transcription.transform) || !transcription.preInput.isEqual(Mock.from(outputTarget))) {
```

Comment:
Why both?

1. `transcription.transform` (and in fact, `transcription` in general) does not provide any indication of deadkey changes.
2. While certainly "bad keyboard design", there's nothing preventing a multitap component of a layer-switch key from manipulating the context.  Should that occur, the correct thing to do is still to rewind the context.

Note that it is _technically_ possible to have altered the context since the base tap... but if the current context state matches the original one anyway, there's no need to perform a reset - we've already done the rewind in one manner or other.

----

We're only safe to bypass the context-rewind mechanism if there truly has been _no_ change to the context whatsoever.  Deadkey changes are context changes, and to handle them requires the some of the special handling newly added by this PR.